### ISSUE TITLE
Deleting scalar list throws error ' violates foreign key constraint'

### ISSUE THREAD

**Issue by Jaszkowic** [prisma/prisma#2621]
I am trying to perform a cascading delete on `User` for the schema below. Note, that `Course` has a scalar list field `colors: [String!]!`. The deletion is working properly when the field `colors` is empty. However, when it is **not** empty, I receive the following error (obtained with `docker-compose logs`:

> [36mpostgres_1  |[0m 2018-06-12 09:28:24.910 UTC [58] ERROR:  update or delete on table "Course" violates foreign key constraint "Course_colors_nodeId_fkey" on table "Course_colors"
[36mpostgres_1  |[0m 2018-06-12 09:28:24.910 UTC [58] DETAIL:  Key (id)=(cjibh0fvk003808019s4j55mk) is still referenced from table "Course_colors".

It seems like the Postgresql connector is not properly  deleting the corresponding entries in the automatically generated table `Course_colors`. 

Or maybe the schema is designed in a wrong way?

```
type User {
    id: ID! @unique,
    createdAt: DateTime!,
    courses: [Course!]! @relation(name: "CoursesForUser", onDelete: CASCADE)
}

type Course {
    id: ID! @unique,
    createdAt: DateTime!,
    colors: [String!]!,
    user: [User!]! @relation(name: "CoursesForUser")
}
```

**Comment by Aetherall** [comment]
Same here, can't delete an user containing a scalar list as a field :/

```gql
type User {
  id: ID! @unique  
  createdAt: DateTime!   
  updatedAt: DateTime!   
  username: String @unique  
  groups: [String!]!
  emails: [Email!]!
  services: [Service!]!
  profile: Profile
}
```
<details>
<summary>The error :</summary>
    <p>

```scala
org.postgresql.util.PSQLException: ERROR: update or delete on table "User" violates foreign key constraint
"User_groups_nodeId_fkey" on table "User_groups"

Detail: Key (id)=(cjibsmf4s02g20a54y930o6a0) is still referenced from table "User_groups".
```

</p></details>

<details>
<summary>Some additionnal logs :</summary>
    <p>

```scala
org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2433)
org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2178)
org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:306)
org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:441)
org.postgresql.jdbc.PgStatement.execute(PgStatement.java:365)
org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:155)
org.postgresql.jdbc.PgPreparedStatement.execute(PgPreparedStatement.java:144)
com.zaxxer.hikari.pool.ProxyPreparedStatement.execute(ProxyPreparedStatement.java:44)
com.zaxxer.hikari.pool.HikariProxyPreparedStatement.execute(HikariProxyPreparedStatement.java)
slick.jdbc.StatementInvoker.results(StatementInvoker.scala:39)
slick.jdbc.StatementInvoker.iteratorTo(StatementInvoker.scala:22)
slick.jdbc.Invoker.first(Invoker.scala:31)
slick.jdbc.Invoker.first$(Invoker.scala:30)
slick.jdbc.StatementInvoker.first(StatementInvoker.scala:16)
slick.jdbc.StreamingInvokerAction$HeadAction.run(StreamingInvokerAction.scala:52)
slick.jdbc.StreamingInvokerAction$HeadAction.run(StreamingInvokerAction.scala:51)
slick.dbio.DBIOAction$$anon$4.$anonfun$run$3(DBIOAction.scala:240)
scala.collection.Iterator.foreach(Iterator.scala:929)
scala.collection.Iterator.foreach$(Iterator.scala:929)
scala.collection.AbstractIterator.foreach(Iterator.scala:1417)
scala.collection.IterableLike.foreach(IterableLike.scala:71)
scala.collection.IterableLike.foreach$(IterableLike.scala:70)
scala.collection.AbstractIterable.foreach(Iterable.scala:54)
slick.dbio.DBIOAction$$anon$4.run(DBIOAction.scala:240)
slick.dbio.DBIOAction$$anon$4.run(DBIOAction.scala:238)
slick.dbio.SynchronousDatabaseAction$FusedAndThenAction.$anonfun$run$4(DBIOAction.scala:534)
slick.dbio.SynchronousDatabaseAction$FusedAndThenAction.$anonfun$run$4$adapted(DBIOAction.scala:534)
scala.collection.Iterator.foreach(Iterator.scala:929)
scala.collection.Iterator.foreach$(Iterator.scala:929)
scala.collection.AbstractIterator.foreach(Iterator.scala:1417)
scala.collection.IterableLike.foreach(IterableLike.scala:71)
scala.collection.IterableLike.foreach$(IterableLike.scala:70)
scala.collection.AbstractIterable.foreach(Iterable.scala:54)
slick.dbio.SynchronousDatabaseAction$FusedAndThenAction.run(DBIOAction.scala:534)
slick.dbio.DBIOAction$$anon$1.$anonfun$run$1(DBIOAction.scala:187)
scala.collection.Iterator.foreach(Iterator.scala:929)
scala.collection.Iterator.foreach$(Iterator.scala:929)
scala.collection.AbstractIterator.foreach(Iterator.scala:1417)
scala.collection.IterableLike.foreach(IterableLike.scala:71)
scala.collection.IterableLike.foreach$(IterableLike.scala:70)
scala.collection.AbstractIterable.foreach(Iterable.scala:54)
slick.dbio.DBIOAction$$anon$1.run(DBIOAction.scala:187)
slick.dbio.DBIOAction$$anon$1.run(DBIOAction.scala:184)
slick.dbio.SynchronousDatabaseAction$FusedAndThenAction.$anonfun$run$4(DBIOAction.scala:534)
slick.dbio.SynchronousDatabaseAction$FusedAndThenAction.$anonfun$run$4$adapted(DBIOAction.scala:534)
scala.collection.Iterator.foreach(Iterator.scala:929)
scala.collection.Iterator.foreach$(Iterator.scala:929)
scala.collection.AbstractIterator.foreach(Iterator.scala:1417)
scala.collection.IterableLike.foreach(IterableLike.scala:71)
scala.collection.IterableLike.foreach$(IterableLike.scala:70)
scala.collection.AbstractIterable.foreach(Iterable.scala:54)
slick.dbio.SynchronousDatabaseAction$FusedAndThenAction.run(DBIOAction.scala:534)
slick.dbio.SynchronousDatabaseAction$$anon$11.run(DBIOAction.scala:571)
slick.dbio.SynchronousDatabaseAction$$anon$6.run(DBIOAction.scala:470)
slick.dbio.SynchronousDatabaseAction$$anon$10.run(DBIOAction.scala:562)
slick.dbio.SynchronousDatabaseAction$$anon$7.run(DBIOAction.scala:487)
slick.basic.BasicBackend$DatabaseDef$$anon$2.liftedTree1$1(BasicBackend.scala:240)
slick.basic.BasicBackend$DatabaseDef$$anon$2.run(BasicBackend.scala:240)
java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
java.lang.Thread.run(Thread.java:748)
```

</p></details>

**Comment by Jaszkowic** [comment]
Any updates on this? Or did someone find a workaround for this?

**Comment by Aetherall** [comment]
I use mysql for now ....

**Comment by Jaszkowic** [comment]
What also works is to update the fields to be empty again before the deletion.

```
mutation {
  updateCourse(
    where: { id: "cjivp6ydh14930707kn0q1ble" }, 
  	data: {
    	colors: {set: []},
  	})
  {
    id
    colors
  },
  
  deleteCourse(
  	where: { id: "cjivp6ydh14930707kn0q1ble" } )
  {
    id
  }
  
}
```

**Comment by marktani** [comment]
Thanks for your additional input! We will add a resolution to the beta channel soon 🙂

**Comment by marktani** [comment]
This was fixed in [this PR](https://github.com/prismagraphql/prisma/pull/2647) and is now available in `1.12-beta`.
